### PR TITLE
Lidar Lite v3 fix for Linux

### PR DESCRIPTION
**Pull request by OXINARF**
This is a fix @tridge did some time ago to fix the Lidar Lite v3 for Linux. If I'm not mistaken in PX4 we always do separate receive and send transfers but that's not the case for Linux.

@patrickpoirier51 on Gitter confirmed this works so it would be nice to get it into master.

### CODE REVIEW

**Comment by patrickpoirier51** [comment]
We need to apply a  ''quick fix'' in order to get the  Lidar V3 initialised  to required settings by forcing the result      v2_hardware = true;  
https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_RangeFinder/AP_RangeFinder_PulsedLightLRF.cpp#L161

**Comment by OXINARF** [comment]
@lucasdemarchi After all, this needs more work, but can you confirm you are OK with the approach to do a double I2C transfer?

File: libraries/AP_HAL_Linux/I2CDevice.cpp [hunk 1]
```diff
@@ -154,6 +154,16 @@ I2CDevice::~I2CDevice()
 bool I2CDevice::transfer(const uint8_t *send, uint32_t send_len,
                          uint8_t *recv, uint32_t recv_len)
 {
+    if (send_len > 0 && recv_len > 0 && _split_transfers) {
+        /*
+          optionally split all I2C transfers. This is needed for some
+          devices (such as LidarLite) that require that STOP only
+          happen with SCL high
+         */
```
Comment (lucasdemarchi) [inline libraries/AP_HAL_Linux/I2CDevice.cpp]: From what I can see on Lidar Lite V3 datasheet (http://static.garmin.com/pumac/LIDAR_Lite_v3_Operation_Manual_and_Technical_Specifications.pdf, page 6) the requirement  is actually that there's a stop condition between the write and read operations. It's like having a situation like in MS5611 in which you first send a command and later read the value from the register.

So... from what I can see this is more a helper method to allow using methods like read_registers().

It's nitpick but I would actually prefer this statement as:

```c++
if (_split_transfers && send_len > 0 && recv_len > 0) {
```

so the **intent** is clearer and the compiler optimizes out the other checks for all other drivers

**Comment by lucasdemarchi** [comment]
Applied, with 2 commits on top to fix the comment.

**Pull request closed without merging**